web/pdf_page_view.js

```diff
@@ -474,10 +474,13 @@ class PDFPageView {
     }
 
     const treeDom = await this.structTreeLayer?.render();
-    if (treeDom && this.canvas && treeDom.parentNode !== this.canvas) {
-      // Pause translation when inserting the structTree in the DOM.
+    if (treeDom) {
       this.l10n.pause();
-      this.canvas.append(treeDom);
+      this.structTreeLayer?.addElementsToTextLayer();
```

Comment:
With this line added here, I do think that we need to *also* apply the following changes; please see the diff below.
That diff shouldn't affect the annotationLayer negatively, given how it's accessing the structTree, but it will ensure that that a *re-created* textLayer cannot ever miss these img-element outlines.
(Given how the viewer currently works it's probably not an issue, but let's avoid a future *intermittent* and hard to find bug here.)

```diff
diff --git a/web/pdf_page_view.js b/web/pdf_page_view.js
index 1005ed29e..e3c154b84 100644
--- a/web/pdf_page_view.js
+++ b/web/pdf_page_view.js
@@ -771,7 +771,7 @@ class PDFPageView {
       this.annotationLayer = null;
       this._annotationCanvasMap = null;
     }
-    if (this.structTreeLayer && !(this.textLayer || this.annotationLayer)) {
+    if (this.structTreeLayer && !this.textLayer) {
       this.structTreeLayer = null;
     }
     if (
```